pkg/etcdcli/health.go

```diff
@@ -0,0 +1,50 @@
+package etcdcli
+
+import (
+	"go.etcd.io/etcd/etcdserver/etcdserverpb"
+)
+
+type healthCheck struct {
+	Member  *etcdserverpb.Member
+	Health  bool
+	Started bool
+	Took    string
+	Error   string
+}
+
+type memberHealth struct {
+	Check []healthCheck
+}
```

Comment:
Another way to represent this might be...

```go
type memberHealth []healthCheck

func (m memberHealth) Healthy() []*etcdserverpb.Member {
  // filter m by healthy
}
func (m memberHealth) Unhealthy() []*etcdserverpb.Member {
  // filter m by unhealthy
}
// ... etc
```

One possible advantage is clearer caller semantics compared to multiple return values

Reply:
yep I like that